### PR TITLE
1715 stanage going live update choosing appropriate compute resources

### PR DESCRIPTION
**Pull request by nikishe**
@Carldkennedy @jkwmoore going to be committing "Choosing appropriate compute resources" change, Theres a chance it might break once carl does his commit , but I will make any necessary changes when Carls update  is done. Pay extra attention to the "Cluster choice" section as I had to reword it

### CODE REVIEW

**Comment by jkwmoore** [comment]
> @Carldkennedy @jkwmoore going to be committing "Choosing appropriate compute resources" change, Theres a chance it might break once carl does his commit , but I will make any necessary changes when Carls update is done. Pay extra attention to the "Cluster choice" section as I had to reword it

Did you mean when merging into main there might be a conflict depending on the merge order of the other PRs?

File: hpc/Choosing-appropriate-resources.rst [hunk 1]
```diff
@@ -34,11 +34,11 @@ This does not mean that you can set extremely large values for these resource re
 Cluster choice
 ==============================
 
-It is also important to note that the Sheffield HPC clusters have been designed to fulfil different purposes. ShARC is for the most part a *capability* cluster designed to run larger compute jobs that will use multiple nodes. Bessemer is a *capacity* cluster designed to run smaller compute jobs which will fit on a single node. In addition, Bessemer has more modern faster CPUs but does not have a /data filestore.
+Stanage and Bessemer have newer CPUs with more modern features.
 
-With this in mind, you should prioritize putting smaller core count jobs onto Bessemer and massively parallel jobs onto ShARC (while utilizing a form of :ref:`MPI <parallel_MPI>`).
+You should prioritize putting smaller core count jobs onto Stannage or Bessemer and massively parallel jobs onto Stanage or ShARC(while utilizing a form of :ref:`MPI <parallel_MPI>`).
```
Comment (jkwmoore) [inline hpc/Choosing-appropriate-resources.rst:39]:
```suggestion
You should prioritize putting smaller core count jobs onto Stanage or Bessemer and massively parallel jobs onto Stanage or ShARC(while utilizing a form of :ref:`MPI <parallel_MPI>`).
```

File: hpc/Choosing-appropriate-resources.rst [hunk 2]
```diff
@@ -34,11 +34,11 @@ This does not mean that you can set extremely large values for these resource re
 Cluster choice
 ==============================
 
-It is also important to note that the Sheffield HPC clusters have been designed to fulfil different purposes. ShARC is for the most part a *capability* cluster designed to run larger compute jobs that will use multiple nodes. Bessemer is a *capacity* cluster designed to run smaller compute jobs which will fit on a single node. In addition, Bessemer has more modern faster CPUs but does not have a /data filestore.
```
Comment (jkwmoore) [inline hpc/Choosing-appropriate-resources.rst:37]: Please add this back and modify it to include Stanage. Stanage is intended to be doing the same role as ShARC but is newer.

File: hpc/Choosing-appropriate-resources.rst [hunk 3]
```diff
@@ -34,11 +34,11 @@ This does not mean that you can set extremely large values for these resource re
...
+Stanage and Bessemer have newer CPUs with more modern features.
```
Comment (jkwmoore) [inline hpc/Choosing-appropriate-resources.rst:37]: Can we clarify this also, Stanage is quite a bit newer than Bessemer too.

File: hpc/Choosing-appropriate-resources.rst [hunk 4]
```diff
@@ -96,11 +96,11 @@ In order to determine your CPU requirements, you should investigate if your prog
 
 If your job / program supports multiple cores, you need to assess whether it supports SMP (symmetric multiprocessing) where you can only use CPUs on 1 node or MPI (message passing interface) where you can access as many nodes, CPUs and cores as are available.
 
-For SMP only type parallel processing jobs: you can use a maximum of 16 cores on ShARC and 40 cores on Bessemer. Ideally you should use Bessemer as you can not only access more cores, you are using more modern cores.
+For SMP only type parallel processing jobs: you can use a maximum of 16 cores on ShARC and 40 cores on Bessemer. Ideally you should use Stanage or Bessemer as you can not only access more cores, you are using more modern cores.
```
Comment (jkwmoore) [inline hpc/Choosing-appropriate-resources.rst:99]: Please rework this line to include the maximum SMP core count (64) for normal Stanage nodes.

File: hpc/Choosing-appropriate-resources.rst [hunk 5]
```diff
@@ -96,11 +96,11 @@ In order to determine your CPU requirements, you should investigate if your prog
...
 
 For multiple node MPI type parallel processing jobs: these can only run on ShARC and although you can access as many cores as are available you must consider how long a job will take to queue waiting for resources compared the the decrease in time for the job to complete computation.
```
Comment (jkwmoore) [inline hpc/Choosing-appropriate-resources.rst:101]: This line is incorrect. Multiple node MPI type jobs can run on Stanage.

**Comment by jkwmoore** [comment]
Can we remove any references to the **timeused** command as I do not believe this is actually available on any of the clusters.

Ln 69-77 can be removed from `sheffield_hpc/hpc/Choosing-appropriate-resources.rst`

File: referenceinfo/imports/scheduler/cpu_allocation_limits_table_import.rst [hunk 6]
```diff
@@ -6,14 +6,19 @@
...
-     - Submission Argument
+     - Submission Argument 
 
-   * - SGE (ShARC)
-     - 1 / 1 / ~1536 (MPI), 16 (SMP)
-     - 1 / 1 / ~1536 (MPI), 16 (SMP)
-     - ``-pe <env> <nn>``
+   * - SLURM (Stanage)
+     - 1 / 1 / 64
+     - 1 / 1 / 64
```
Comment (jkwmoore) [inline referenceinfo/imports/scheduler/cpu_allocation_limits_table_import.rst:13]: These need correcting to follow the same format as ShARC. The number of cores per node is 64 / total cluster cores for MPI should be on the Stanage specs page.

File: referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst [hunk 7]
```diff
@@ -1,25 +1,36 @@
...
+   * - SLURM (Stanage)
      - 256 GB
-     - 2 GB / 64 GB
-     - 2 GB / 64 GB (SMP) ~6144 GB (MPI)
-     - **Per core basis** ``-l rmem=<nn>``
+     - 1TB 
+     - 2TB
+     - 2 GB / (256 GB/ 1TB / 2TB) 
+     - 2 GB / (256 GB/ 1TB / 2TB) 
+     - **Per job basis** ``--mem=<xxG>``
```
Comment (jkwmoore) [inline referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst:20]:
```suggestion
     - **Per job basis** ``--mem=<nn>``
```

File: referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst [hunk 8]
```diff
@@ -1,25 +1,36 @@
...
-   * - SGE (ShARC)
-     - 64 GB
+   * - SLURM (Stanage)
      - 256 GB
-     - 2 GB / 64 GB
-     - 2 GB / 64 GB (SMP) ~6144 GB (MPI)
-     - **Per core basis** ``-l rmem=<nn>``
+     - 1TB 
+     - 2TB
+     - 2 GB / (256 GB/ 1TB / 2TB)
```
Comment (jkwmoore) [inline referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst:18]:
```suggestion
     - 2 GB / 256 GB
```

File: referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst [hunk 9]
```diff
@@ -1,25 +1,36 @@
...
-     - 64 GB
+   * - SLURM (Stanage)
      - 256 GB
-     - 2 GB / 64 GB
-     - 2 GB / 64 GB (SMP) ~6144 GB (MPI)
-     - **Per core basis** ``-l rmem=<nn>``
+     - 1TB 
+     - 2TB
+     - 2 GB / (256 GB/ 1TB / 2TB) 
+     - 2 GB / (256 GB/ 1TB / 2TB)
```
Comment (jkwmoore) [inline referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst:19]: This isn't correct as this again needs to reflect the SMP amount and MPI amounts available based on single node vs all nodes memory.

File: referenceinfo/imports/scheduler/memory_used_qstat.rst
Comment (jkwmoore) [inline referenceinfo/imports/scheduler/memory_used_qstat.rst:1]: Can this file and its import be renamed to be more descriptive / accurate - this import does not only reference qstat.

File: referenceinfo/scheduler/SGE/sge_parallel_environments.rst [hunk 10]
```diff
@@ -2,7 +2,7 @@
 ShARC Parallel Environments
 ---------------------------
 
-The available :ref:`parallel environments <parallel>` for ShARC only (Bessemer only has the SMP environment which is set by default) can be found below:
+The available :ref:`parallel environments <parallel>` for ShARC only (Only Stanage and Bessemer have the SMP environment which is set by default) can be found below:
```
Comment (jkwmoore) [inline referenceinfo/scheduler/SGE/sge_parallel_environments.rst:5]: This is wrong as all clusters have the ability to run SMP jobs, but only ShARC has an SGE parallel environment called smp. Suggest we remove references to Stanage and Bessemer, or fully explain the difference.

**Comment by jkwmoore** [comment]
@willfurnass Can you roll through checking the memory values for Stanage - I don't have the (slightly reduced) single node values to hand/memory.

File: referenceinfo/imports/scheduler/memory_used_qstat.rst
Comment (nikishe) [reply referenceinfo/imports/scheduler/memory_used_qstat.rst:1]: updated to memory used commands

File: referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst (hunk 9 above)
Comment (nikishe) [reply referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst]: updated to reflect this

**Comment by nikishe** [comment]
@jkwmoore It was more referring to a broken link , but that's an easy fix for after Carls update

File: referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst (hunk 9 above)
Comment (jkwmoore) [reply referenceinfo/imports/scheduler/memory_allocation_limits_table_import.rst]: I don't see this amendment. Line 19 should be reflecting 3 values - default, SMP and MPI.

File: referenceinfo/imports/scheduler/cpu_allocation_limits_table_import.rst [hunk 11]
```diff
@@ -6,14 +6,19 @@
...
-     - Submission Argument
+     - Submission Argument 
 
-   * - SGE (ShARC)
-     - 1 / 1 / ~1536 (MPI), 16 (SMP)
-     - 1 / 1 / ~1536 (MPI), 16 (SMP)
-     - ``-pe <env> <nn>``
+   * - SLURM (Stanage)
+     - 1 / 1 /  ~9728 (MPI), 64 (SMP)
+     - 1 / 1 /  ~9728 (MPI), 64 (SMP)
```
[the remark this answers is not in the document]
Comment (nikishe) [reply referenceinfo/imports/scheduler/cpu_allocation_limits_table_import.rst:13]: Only did the calculations based on the 152 general cpu nodes. Was assuming the large and very large can not be mixed and match into the pool. I take it this assumption was wrong

[the remark this answers is not in the document]
Comment (nikishe) [reply referenceinfo/imports/scheduler/cpu_allocation_limits_table_import.rst:13]: updated to 11264

**Comment by nikishe** [comment]
Fixed the conflicts

File: hpc/Choosing-appropriate-resources.rst [hunk 12]
```diff
@@ -34,11 +34,18 @@ This does not mean that you can set extremely large values for these resource re
...
+We have three cluster choices listed below for you to choose from:
 
-With this in mind, you should prioritize putting smaller core count jobs onto Bessemer and massively parallel jobs onto ShARC (while utilizing a form of :ref:`MPI <parallel_MPI>`).
+* Stanage (Our newest and most powerful yet, launched in March 2023)
+* Bessemer (Launched in 2018)
+* ShARC (Launched in 2017)
 
-The specifications for each cluster are detailed for  ShARC here :ref:`sharc-specs` and Bessemer here. :ref:`bessemer-specs`
+It is also important to note that the Sheffield HPC clusters have been designed to fulfil different purposes. Stanage and ShARC are for the most part *capability* clusters designed to run larger compute jobs that will use multiple nodes. Bessemer is a *capacity* cluster designed to run smaller compute jobs which will fit on a single node. In addition, Stanage and Bessemer have newer CPUs with more modern features. Bessemer but does not have a /data filestore.
```
Comment (Carldkennedy) [inline hpc/Choosing-appropriate-resources.rst:43]: Stanage doesn't have  /data either

**Review by Carldkennedy — COMMENTED** [review]
LGTM with one comment

File: hpc/Choosing-appropriate-resources.rst (hunk 12 above)
Comment (nikishe) [reply hpc/Choosing-appropriate-resources.rst]: updated

**Pull request merged**